packages/lib-classifier/src/plugins/tasks/DrawingTask/models/DrawingAnnotation.spec.js

```diff
@@ -1,19 +1,187 @@
+import { types } from 'mobx-state-tree'
+import DrawingTask from './DrawingTask'
 import DrawingAnnotation from './DrawingAnnotation'
 import { Point } from '@plugins/drawingTools/models/marks'
 
-const point = Point.create({ id: 'mockAnnotation', frame: 0, toolIndex: 0, toolType: 'point', x: 100, y: 150 })
+describe('Model > DrawingAnnotation', function () {
+  const point = Point.create({ id: 'mockAnnotation', frame: 0, toolIndex: 0, toolType: 'point', x: 100, y: 150 })
 
-const drawingAnnotationSnapshot = {
-  id: 'drawing1',
-  task: 'T0',
-  taskType: 'drawing',
-  value: [ point.id ]
-}
 
-describe('Model > DrawingAnnotation', function () {
+  const drawingAnnotationSnapshot = {
+    id: 'drawing1',
+    task: 'T0',
+    taskType: 'drawing',
+    value: [ point.id ]
+  }
+
   it('should exist', function () {
     const drawingAnnotation = DrawingAnnotation.create(drawingAnnotationSnapshot)
     expect(drawingAnnotation).to.exist()
     expect(drawingAnnotation).to.be.an('object')
   })
+
+  describe('annotation snapshots', function () {
+    describe('without subtasks', function () {
+      let snapshot
+
+      before(function () {
+        const drawingTask = DrawingTask.create({
+          instruction: 'draw things!',
+          taskKey: 'T0',
+          tools: [
+            {
+              type: 'point'
+            }
+          ],
+          type: 'drawing'
+        })
+        const pointTool = drawingTask.tools[0]
+        const point = pointTool.createMark({
+          x: 50,
+          y: 100
+        })
+        const annotation = DrawingAnnotation.create({
+          id: 'drawing1',
+          task: 'T0',
+          taskType: 'drawing',
+          value: [ point ]
+        })
+        const mockStore = types.model('MockStore', {
```

Comment:
We should be careful with mocked parent stores. I ran into a lot of trouble with mocks like these in tests back when I updating the classifier to MST 3.11.0 a few months ago. I'd prefer an actual integration test with the real parents.

Reply:
I don’t think the drawing annotation does any type conversion itself. ADR 25 says it should flatten the annotations array but I think that’s all that needs to be tested here. 

Other components convert maps to arrays but they have their own tests.

Reply:
Just to be clear, this is testing the drawing annotation in isolation. An annotation on its own will throw invalid reference errors, so we test with an annotation and its task.

Integration tests could mock a `ClassificationStore` and a workflow (`WorkflowStepsStore`?), run through annotations for some mock task then complete the classification. That's outside the scope of the annotation tests but `TaskNavButtons` has code that does this: https://github.com/zooniverse/front-end-monorepo/blob/f4f704fea6f24009270d6df92e0887b2e882067f/packages/lib-classifier/src/components/Classifier/components/TaskArea/components/Tasks/components/TaskNavButtons/TaskNavButtonsContainer.spec.js#L21-L67

Reply:
My original comment here doesn't have to do with type conversion, so I'm not sure what to say about that.

I really don't think we should use mocks for stores at all. I completely disagree with that approach based on the experience I had upgrading to MST 3.11.0 and the massive test rewrite I had to do to figure out what was going wrong (the mock stores didn't reflect the reality of how stores were being used), the comments on how to do tests from the core maintainers of MST, and general reading on when to use mocks in testing and when not to. I've been moving away from doing this whenever I write store tests.

I won't block this PR over this, but I really think don't think we should be testing like this.

Reply:
I'm not really sure how to test references any other way. The referenced model has to exist in the same tree somewhere. In this case that's the marks that make up the annotation.

Reply:
>I really think don't think we should be testing like this.

That's fine but if you could share what you've been reading, and what you've learnt then the rest of us can learn from it too. At the very least, some guidance on how to test the models would be useful. Here, I have no idea how to test a view that resolves identifiers elsewhere in the tree. All I've got to go on is fairly sketchy advice from the MST forums.
https://spectrum.chat/mobx-state-tree/general/getsnapshot-with-reference-field~43a6e2d2-9655-4571-b599-9cac4ee9c94d
https://spectrum.chat/mobx-state-tree/general/getsnapshot-with-resolved-references~ba70fa0a-9c45-4cee-a2e0-5213d01132fc

Reply:
I've shared this before, and this is a good place to start: https://github.com/mobxjs/mobx-state-tree/issues/1404 and specifically this comment: https://github.com/mobxjs/mobx-state-tree/issues/1404#issuecomment-544453565